scipy/interpolate/_isotonic.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from scipy.optimize import isotonic_regression
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+__all__ = ["IsotonicInterpolator"]
+
+
+# TODO: Should this inherit from _Interpolator1D?
+class IsotonicInterpolator():
+    """Nonparametric isotonic interpolation via PAVA.
+
+    This class provides nonparametric monotonically increasing interpolation
+    via the pool adjacent violators algorithm (PAVA).
+
+    This interpolator first sorts the values of ``y`` according to the order
+    given by ``x``. It then fits the isotonic regression, see
+    ``scipy.optimize.isotonic_regression`` for details. Afterwards it
+    constructs a linear interpolation of the solution of the isotonic
+    regression.
```

Comment:
The docstring certainly needs references + an short explanation of the algorithm + an example.

Reply:
Example yes. But is it enough to link to :func:`scipy.optimize.isotonic_regression`?

Reply:
Depends. 
If we remove sorting, then what is left. The fact that there is not much to say other than "delegates to optimize.isotonic_regression" strongly hints that this whole class is extraneous :-). If there is something in scipy.interpolate, it can and should of course like to whatever implements the algorithm; a short sentence with several key words and a clear message for details see <link to optimize> would be helpful.

Speaking of explanations, I think there's a need for some more so that it's not just "here's a black box, if you care what's inside, go to the original paper", but that can wait until a later iteration.